app/controllers/tasks_controller.rb

```diff
@@ -26,6 +26,31 @@ def index
     @tasks = Task.all
   end
 
+  def edit
+    @task = Task.find_by(requester_id: current_user.id)
+    @task = Task.find(params[:id])
+    @task_categories = TaskCategory.all
+  end
```

Comment:
As I see it, only one task can have the ID of 30. Also, you kind of "throw away" the assignment of @task done on line 30 when going to line 31.
We can have a chat on SH to go over.